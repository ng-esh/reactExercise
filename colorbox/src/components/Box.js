import React from "react";
import "../styles/Box.css";

function Box({ id, width, height, backgroundColor, removeBox }) {
  return (
    <div className="box-container">
      <div 
        className="box"
        style={{ 
          width: `${width}rem`, 
          height: `${height}rem`, 
          backgroundColor 
        }} 
      />
      <button className="remove-btn" onClick={() => removeBox(id)}>X</button>
    </div>
  );
}

export default Box;
