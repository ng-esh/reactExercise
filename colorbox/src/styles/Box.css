.box-container {
    display: flex;
    align-items: center;
    gap: 0.5rem;
    margin: 0.5rem;
  }
  
  .box {
    display: inline-block;
    border: 2px solid black;
  }
  
  .remove-btn {
    background: red;
    color: white;
    border: none;
    cursor: pointer;
    padding: 0.5rem;
    font-size: 1rem;
    font-weight: bold;
  }
  
  .remove-btn:hover {
    background: darkred;
  }
  