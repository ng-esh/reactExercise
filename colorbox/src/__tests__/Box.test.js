import { render, fireEvent } from "@testing-library/react";
import Box from "../components/Box.js";

test("renders without crashing", () => {
  render(<Box width={100} height={100} backgroundColor="blue" />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<Box width={100} height={100} backgroundColor="blue" />);
  expect(asFragment()).toMatchSnapshot();
});

test("calls removeBox function when X button is clicked", () => {
  const removeBox = jest.fn();
  const { getByText } = render(<Box id="123" width={100} height={100} backgroundColor="blue" removeBox={removeBox} />);
  
  fireEvent.click(getByText("X"));
  expect(removeBox).toHaveBeenCalledWith("123");
});
