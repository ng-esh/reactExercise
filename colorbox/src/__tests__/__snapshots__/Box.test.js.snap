// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="box-container"
  >
    <div
      class="box"
      style="width: 100rem; height: 100rem; background-color: blue;"
    />
    <button
      class="remove-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
